services/frontend-service/src/ui/components/ReleaseCard/ReleaseCard.tsx

```diff
@@ -71,51 +71,78 @@ const rolloutStatusPriority = [
     RolloutStatus.ROLLOUT_STATUS_SUCCESFUL,
 ];
 
+const getRolloutStatusPriority = (status: RolloutStatus): number => {
+    const idx = rolloutStatusPriority.indexOf(status);
+    if (idx === -1) {
+        return rolloutStatusPriority.length;
+    }
+    return idx;
+};
+
+type DeploymentStatus = {
+    environmentGroup: string;
+    rolloutStatus: RolloutStatus;
+};
+
+const getAppRolloutStatus = (
+    status: StreamStatusResponse | undefined,
+    deployedVersion: number | undefined
+): RolloutStatus => {
+    if (status === undefined) {
+        // The status is completly unknown. Either the app is just new or rollout service is not responding.
+        return RolloutStatus.ROLLOUT_STATUS_UNKNOWN;
+    }
+    if (status.rolloutStatus === RolloutStatus.ROLLOUT_STATUS_SUCCESFUL && status.version !== deployedVersion) {
+        // The rollout service might be sligthly behind the UI.
+        return RolloutStatus.ROLLOUT_STATUS_PENDING;
+    }
+    return status.rolloutStatus;
+};
+
 const calculateDeploymentStatus = (
     app: string,
     deployedAt: EnvironmentGroupExtended[],
     rolloutEnabled: boolean,
     rolloutStatus: RolloutStatusApplication
-): [Array<StreamStatusResponse>, StreamStatusResponse?] => {
+): [Array<DeploymentStatus>, RolloutStatus?] => {
     if (!rolloutEnabled) {
         return [[], undefined];
     }
-    const rolloutEnvs = deployedAt.flatMap((envGroup) =>
-        envGroup.environments.map((env) => {
-            const status = rolloutStatus[env.name];
-            if (!status) {
-                // The status is completly unknown. Either the app is just new or rollout service is not responding.
-                return {
-                    environment: env.name,
-                    rolloutStatus: RolloutStatus.ROLLOUT_STATUS_UNKNOWN,
-                    version: 0,
-                    application: app,
-                };
+    const rolloutEnvs = deployedAt.map((envGroup) => {
```

Comment:
rolloutEnvs is closer to a list of groups, rather than env:

```suggestion
    const rolloutEnvGroups = deployedAt.map((envGroup) => {
```